src/inmanta/parser/plyInmantaParser.py

```diff
@@ -523,7 +523,7 @@ def p_multi_3(p: YaccProduction) -> None:
 
 def p_multi_4(p: YaccProduction) -> None:
     "multi : '['  ':' INT ']' "
-    p[0] = (None, p[3])
+    p[0] = (0, p[3])
```

Comment:
@edvgui FYI, this was the actual fix, the rest of this PR are just typing fixes to clean this up.